examples/cosmos/ics20/ics20.qnt

```diff
@@ -146,90 +219,266 @@ module ics20Test {
-    pure val connectedChains = channelTopology.get(chain).keys()
+  pure val channelCounterparties: str -> Channel -> Channel = CHANNEL_TOPOLOGY.keys().mapBy(chain => {
+    pure val connectedChains = CHANNEL_TOPOLOGY.get(chain).keys()
     connectedChains.map(counterpartyChain => {
-      pure val localChannel = channelTopology.get(chain).get(counterpartyChain)
-      pure val counterpartyChannel = channelTopology.get(counterpartyChain).get(chain)
+      pure val localChannel = CHANNEL_TOPOLOGY.get(chain).get(counterpartyChain)
+      pure val counterpartyChannel = CHANNEL_TOPOLOGY.get(counterpartyChain).get(chain)
       (localChannel, counterpartyChannel)
     }).setToMap()
   })
 
+  /// Return true iff there is a channel from `sourceChain` to `destChain` defined in `CHANNEL_TOPOLOGY`
+  /// that corresponds to the channel endpoints of `packet`.
+  pure def isTravelingBetween(packet: Packet, sourceChain: str, destChain: str): bool = all {
+    // there is a channel between `sourceChain` and `destChain`
+    CHANNEL_TOPOLOGY.keys().contains(sourceChain),
+    CHANNEL_TOPOLOGY.get(sourceChain).keys().contains(destChain),
+    // packet endpoints correspond to channel ends
+    channelCounterparties.get(sourceChain).keys().contains(packet.sourceChannel),
+    channelCounterparties.get(sourceChain).get(packet.sourceChannel) == packet.destChannel
+  }
+
   action init = {
-    chainStates' = channelTopology.keys().mapBy(chain => {
+    chainStates' = CHANNEL_TOPOLOGY.keys().mapBy(chain => {
       // All accounts are empty, except for Alice in chain A who has 100 atoms
-      bank: if (chain == "A") Map("alice" -> Map(toDenom("atom") -> 100)) else Map(),
+      bank: if (chain == "A") Map("alice" -> Map(ATOM -> 100)) else Map(),
       channels: channelCounterparties.get(chain),
-      channelEscrowAddresses: channelCounterparties.get(chain).keys().mapBy(_ => "escrow_account"),
-      outPackets: Set()
+      channelEscrowAddresses: channelCounterparties.get(chain).keys().mapBy(_ => ESCROW_ACCOUNT),
+      outPackets: Set(),
+      receivedButUnacknowledgedPackets: Set(),
+      inAcknowledgements: Set()
     })
   }
 
-  /// Receives a packet sent from `sourceChain` to `destChain`
-  action receivePacket(sourceChain: str, destChain: str): bool = {
+  /// Send a packet from `sourceChain` to `destChain`, indicating a transfer
+  /// of `amount`-many tokens of `denom` from `sender` to `receiver`.
+  action sendPacket(sourceChain: str, destChain: str,
+                    denom: DenomTrace, amount: UINT256,
+                    sender: Address, receiver: Address): bool = all {
+    // (1) Pre-condition:
+    // ICS 4: "a channel is a pipeline [...] between specific modules on **separate** blockchains"
+    sourceChain != destChain,
+    // there is a channel between `sourceChain` and `destChain`
+    CHANNEL_TOPOLOGY.keys().contains(sourceChain),
+    CHANNEL_TOPOLOGY.get(sourceChain).keys().contains(destChain),
```

Comment:
I really like `isTravelingBetween`. Can we also have a `hasChannelBetween` and call
`CHANNEL_TOPOLOGY.hasChannelBetween(sourceChain, destChain)`?